### DISCUSSION TITLE
Random seed for Chem.MolToSmiles(mol, doRandom=True)

### DISCUSSION THREAD

**Discussion by daanjiskoot** [rdkit/rdkit#6489] [Ideas] [answered]
Dear all, 

I am looking for help with the doRandom option in Chem.MolToSmiles. This function makes sure that SMILES that are returned are random which is a necessity for the functionality of my code. I also need to set the seed, however, random.seed(42) and np.random.seed(42) do not seem to effect this function. As far as I'm aware there is not yet an option to set the seed using rdkits randomSeed() parameter, which is is a possibility in, for instance, the AllChem.EmbedMolecule function. 

Would anybody know of a way to fix the seed when using doRandom=True? Or is there a possibility that randomSeed() can be added to this function? 

The code looks like this:
`Chem.MolToSmiles(mol, canonical=False, doRandom=True, isomericSmiles=True, kekuleSmiles=True)`

Any help would be greatly appreciated!

Kind regards,
Daan

**Comment by greglandrum** [comment, marked as answer]
Hi @daanjiskoot, at the moment it's not possible to control the random number seed which is used (or to reset it).
The RNG is seeded at program start, so the results that the RDKit generates should be reproducible across runs, as this shows:
```
$ cat blah.py
from rdkit import Chem
m = Chem.MolFromSmiles('OCCC(F)CC(CCF)CC(O)(CCCCCl)O')
print(len(set([Chem.MolToSmiles(m,canonical=False,doRandom=True) for i in range(100)])))
(py311_rdkit) 
glandrum@ferret MINGW64 ~
$ python blah.py
87
(py311_rdkit) 
glandrum@ferret MINGW64 ~
$ python blah.py
87
(py311_rdkit) 
glandrum@ferret MINGW64 ~
$ python blah.py
87
```
but the RNG seed it not reset between calls to `MolToSmiles()`, so you get different results if you call it more than once on the same molecule in the same python session:
```
(py311_rdkit) 
glandrum@ferret MINGW64 ~
$ winpty python -i blah.py
87
>>> print(len(set([Chem.MolToSmiles(m,canonical=False,doRandom=True) for i in range(100)])))
89
>>> print(len(set([Chem.MolToSmiles(m,canonical=False,doRandom=True) for i in range(100)])))
84
>>> print(len(set([Chem.MolToSmiles(m,canonical=False,doRandom=True) for i in range(100)])))
84
>>> print(len(set([Chem.MolToSmiles(m,canonical=False,doRandom=True) for i in range(100)])))
90
```
(These results were generated on my windows laptop, but they should be the same on linux or the mac)

I believe this behavior gives you the option for reproducibility when you need it without further complicating the API for `MolToSmiles()`. Does that seem right to you?

**Reply by daanjiskoot** [reply]
Hi @greglandrum, this answers my question and works similar from my end! Thank you for your quick and elaborate response!

**Comment by Volowan** [comment]
Hello, I had the same problem and the use of:

`Chem.MolToRandomSmilesVect(mol, 1,  isomericSmiles=False, kekuleSmiles=True, randomSeed=np.random.randint(0, 1000))[0]`

Instead of `Chem.MolToSmiles(...,doRandom=True)` seem to have resolved my problem :)

(I think "isomericSmiles" and "kekuleSmiles" are not mandatory but they were needed in my case)

Not sure if there will be people out there with the same problem but as I needed this without restarting the script if possible, I thought this might be useful to some other people.

Best and thank you so much for the great work with rdkit,

Theo
